addons/point_of_sale/static/src/js/Screens/TicketScreen/TicketScreen.js

```diff
@@ -257,6 +254,14 @@ class TicketScreen extends IndependentToOrderScreen {
 
         this._onCloseScreen();
     }
+    _setDestinationOrder(order, partner) {
+        if (order && partner === this.props.destinationOrder.get_partner() && !this.env.pos.doNotAllowRefundAndSales()) {
+            return order;
+        } else if(this.env.pos.get_order() && !this.env.pos.get_order().orderlines.length) {
+            return this.env.pos.get_order();
+        }
+        return this.env.pos.add_new_order({ silent: true });
```

Comment:
Maybe use what has been done before `this._getEmptyOrder(partner);` instead of checking if an empty order already exist as this order might be linked to another partner and you should link the correct partner to it.

Reply:
Also check `this.env.pos.get_order().orderlines.length > 0` instead of just `this.env.pos.get_order().orderlines.length` or clarity reasons